File: src/FoodOrderSystem.App/ClientApp/src/app/order/order.model.ts
import {CustomerInfoModel} from './customer-info.model';
import {CartInfoModel} from './cart-info.model';
import {PaymentMethodModel} from '../payment-method/payment-method.model';

export class OrderModel {
  public id: string;
  public customerInfo: CustomerInfoModel;
  public cartInfo: CartInfoModel;
  public comments: string;
  public paymentMethodId: string;
  public paymentMethod: PaymentMethodModel;
  public valueOfOrder: number;
  public costs: number;
  public totalPrice: number;
}

File: src/Gastromio.Persistence.MongoDB/OrderModel.cs
using System;

namespace Gastromio.Persistence.MongoDB
{
    public class OrderModel
    {
        public Guid Id { get; set; }

        public CustomerInfoModel CustomerInfo { get; set; }

        public CartInfoModel CartInfo { get; set; }

        public string Comments { get; set; }

        public Guid PaymentMethodId { get; set; }

        public string PaymentMethodName { get; set; }

        public string PaymentMethodDescription { get; set; }

        public double Costs { get; set; }

        public double TotalPrice { get; set; }

        public DateTime? ServiceTime { get; set; }

        public NotificationInfoModel CustomerNotificationInfo { get; set; }

        public NotificationInfoModel RestaurantNotificationInfo { get; set; }

        public NotificationInfoModel RestaurantMobileNotificationInfo { get; set; }

        public DateTime CreatedOn { get; set; }

        public DateTime? UpdatedOn { get; set; }

        public Guid? UpdatedBy { get; set; }
    }
}
